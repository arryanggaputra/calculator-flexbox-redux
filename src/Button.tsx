import React from 'react'
import {
  TouchableOpacity,
  Text,
  View,
  StyleProp,
  ViewStyle
} from 'react-native'
import styles from './styles'

interface ButtonProps {
  label: string
  onPress: () => void
  style?: StyleProp<ViewStyle>
}

function Button(props: ButtonProps) {
  return (
    <TouchableOpacity
      onPress={props.onPress}
      style={[styles.buttonArea, props.style]}>
      <View style={styles.button}>
        <Text style={styles.buttonLabel}>{props.label}</Text>
      </View>
    </TouchableOpacity>
  )
}

export default React.memo(
  Button,
  (prevProps, nextProps) => prevProps.label === nextProps.label
)
