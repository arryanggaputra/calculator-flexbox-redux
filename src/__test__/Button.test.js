import React from 'react'
import renderer from 'react-test-renderer'
import jest from 'jest-mock'
import Button from '../Button'
import { TouchableOpacity } from 'react-native'

describe('<Button />', () => {
  it('it should rendered', () => {
    let component = renderer.create(<Button />)
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('it should render correct label', () => {
    let component = renderer.create(<Button label="+" />)
    expect(component.root.props.label).toEqual('+')
  })

  it('it should press correct function', () => {
    const myMock = jest.fn()
    let component = renderer.create(<Button onPress={myMock} label="+" />)
    let button = component.root.findByType(TouchableOpacity)
    button.props.onPress()
    expect(myMock).toHaveBeenCalled()
  })
})
