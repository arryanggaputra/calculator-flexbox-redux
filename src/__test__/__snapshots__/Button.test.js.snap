// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> it should rendered 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexGrow": 1,
      "opacity": 1,
      "padding": 5,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#dddddd",
        "borderRadius": 10,
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 32,
        }
      }
    />
  </View>
</View>
`;
